multithreading/task4/src/main/java/Main.java

```diff
@@ -0,0 +1,13 @@
+public class Main {
+
+  public static void main(String[] args) throws InterruptedException {
+    BlockingObjectPool pool = new BlockingObjectPool(3);
```

Comment:
This program never ends because the capacity of queue is less than the number of items you try to insert. Therefore, the program waits infinitely.